### PR TITLE
Simplify keeping track of ylim dictionary in _plot_topo

### LINKED ISSUE #12723: in mne.viz.plot_evoked_topo, the ylim parameter breaks when grads become before mags in the data
### Description of the problem

In our new MegIn system, the magnetometers come before the gradiometers:

```python
>>> evoked.ch_names
... ['MEG0111',
...  'MEG0112',
...  'MEG0113',
...  'MEG0121',
...  'MEG0122',
...  'MEG0123',
...  ...
...]
```

this causes `mne.viz.plot_evoked_topo` to break, because the order of the channel types is fixed in the handling of `ylim`: https://github.com/mne-tools/mne-python/blob/e299f1efb180968aaf2bb10c1807b8a988124ffa/mne/viz/topo.py#L333

From that code it seems that magnetometers are assumed to come first, but somewhere before the ordering is actually reversed, so gradiometers have to come first:

https://github.com/mne-tools/mne-python/blob/e299f1efb180968aaf2bb10c1807b8a988124ffa/mne/viz/topo.py#L1011

### Steps to reproduce

```Python
import mne
ev = mne.read_evokeds("/home/vanvlm1/mne_data/MNE-sample-data/MEG/sample/sample_audvis-ave.fif", condition="Left Auditory").apply_baseline()

# Re-order channels so that mags come before grads.
picks = list()
for i in range(0, 306, 3):
    picks.append(ev.ch_names[i + 2])
    picks.append(ev.ch_names[i + 0])
    picks.append(ev.ch_names[i + 1])
ev = ev.reorder_channels(picks)

# Make grads huge and mags tiny.
mne.viz.plot_evoked_topo(ev, ylim=dict(grad=[-10, 10], mag=[-1000, 1000]))
```

### Link to data

_No response_

### Expected results

mags should be tiny and grads should be huge.

### Actual results

In the plot, grads are tiny and mags are huge. The opposite of what it should be.

Figure_1

### Additional information

```
Platform             Linux-6.5.0-41-generic-x86_64-with-glibc2.35
Python               3.10.13 | packaged by conda-forge | (main, Dec 23 2023, 15:36:39) [GCC 12.3.0]
Executable           /l/vanvlm1/miniforge3/bin/python
CPU                  x86_64 (32 cores)
Memory               62.5 GB

Core
├☒ mne               1.7.0.dev235+ge43b5d897.d20240516 (outdated, release 1.7.1 is available!)
├☑ numpy             1.26.4 (MKL 2022.1-Product with 24 threads)
├☑ scipy             1.12.0
└☑ matplotlib        3.8.3 (backend=QtAgg)

Numerical (optional)
├☑ sklearn           1.4.1.post1
├☑ numba             0.59.0
├☑ nibabel           5.2.1
├☑ nilearn           0.10.3
├☑ dipy              1.9.0
├☑ openmeeg          2.5.7
├☑ pandas            2.2.1
├☑ h5io              0.2.2
├☑ h5py              3.10.0
└☐ unavailable       cupy

Visualization (optional)
├☑ pyvista           0.43.4 (OpenGL 4.5.0 NVIDIA 535.183.01 via NVIDIA GeForce RTX 4090/PCIe/SSE2)
├☑ pyvistaqt         0.11.0
├☑ vtk               9.2.6
├☑ qtpy              2.4.1 (PyQt5=5.15.8)
├☑ pyqtgraph         0.13.4
├☑ mne-qt-browser    0.6.1
├☑ ipywidgets        8.1.2
├☑ trame_client      2.16.3
├☑ trame_server      2.17.2
├☑ trame_vtk         2.8.5
├☑ trame_vuetify     2.4.3
└☐ unavailable       ipympl

Ecosystem (optional)
├☑ eeglabio          0.0.2-4
├☑ mffpy             0.8.0
└☐ unavailable       mne-bids, mne-nirs, mne-features, mne-connectivity, mne-icalabel, mne-bids-pipeline, neo, edfio, pybv

To update to the latest supported release version to get bugfixes and improvements, visit https://mne.tools/stable/install/updating.html
```

### PR DESCRIPTION
**Pull request by wmvanvliet**
fixes #12723

The handling of the `ylim` parameter in `mne.viz.plot_evoked_topo` is currently weirdly converted into a list of lists, but then transposed... This PR changes it to just remain a dictionary and do away with conversions.

### CODE REVIEW

File: mne/viz/topo.py [hunk 1]
```diff
@@ -330,8 +330,7 @@ def _plot_topo(
 
     for ax, ch_idx in my_topo_plot:
         if layout.kind == "Vectorview-all" and ylim is not None:
-            this_type = {"mag": 0, "grad": 1}[channel_type(info, ch_idx)]
-            ylim_ = [v[this_type] if _check_vlim(v) else v for v in ylim]
+            ylim_ = ylim.get(channel_type(info, ch_idx), None)
```
Comment (drammock) [inline mne/viz/topo.py:333]:
```suggestion
            ylim_ = ylim.get(channel_type(info, ch_idx))
```
None is the default default value for `dict.get()`

**Review by drammock — APPROVED** [review]
Changes look reasonable. Can you touch an example/tutorial or two, so we can see the rendered plots?

**Comment by drammock** [comment]
...also needs a changelog entry (or the "no changelog entry needed" label)

**Comment by larsoner** [comment]
Comments addressed, code looks good to me, and examples look unchanged which I suppose is to be expected, thanks @wmvanvliet !

In a follow-up PR if you're motivated you could in theory take a standard-ordered evoked, plot its topo, then reorder to your new system's order, then make sure that the `ylim` you get are the same. But this might be a bit too much work, especially since the topo don't use proper matplotlib parasite axes but rather our own coordinate system manipulations so I doubt it would be straightforward.

**Pull request merged**